gadget/install/partition.go

```diff
@@ -169,6 +169,7 @@ func buildPartitionList(dl *gadget.OnDiskVolume, pv *gadget.LaidOutVolume) (sfdi
 		toBeCreated = append(toBeCreated, gadget.OnDiskStructure{
 			LaidOutStructure: p,
 			Node:             node,
+			DiskIndex:        pIndex,
```

Comment:
we should probably change one of the comments here to mention that pIndex is 1-based? also we skipping things that are not IsPartition, does it matter for DiskIndex correctness?

Reply:
Yes I added some comments - it is correct and necessary that we skip structures that are not partitions to have a correct index here.